### ISSUE TITLE
Conflicting Peer Dependencies

### ISSUE THREAD

**Issue by youngbloodcyb** [vercel-labs/ai#271]
Error when running `npm i` with the AI SDK installed (version `2.1.14`). Conflicting peer dependency versions between `sswr` package and `ai`. `ai` says Svelte must be version 4 or higher, whereas `sswr` is saying version 3.59.2 or higher. 

I'm getting this error using Next 13.4.7. Ironically I'm not using Svelte at all, but it is causing a dependency issue. Curious to see if anyone else is getting this same issue. Could be something I'm doing or something I can resolve.

I included I full log file below.

```'Log files:
/Users/user/.npm/_logs/2023-07-03T16_14_25_130Z-debug-0.log

# npm resolution error report

While resolving: ai@2.1.14
Found: svelte@3.59.2
node_modules/svelte
  peer svelte@"^3.29.0" from sswr@1.10.0
  node_modules/sswr
    sswr@"^1.10.0" from ai@2.1.14
    node_modules/ai
      ai@"^2.1.14" from the root project

Could not resolve dependency:
peerOptional svelte@"^4.0.0" from ai@2.1.14
node_modules/ai
  ai@"^2.1.14" from the root project

Conflicting peer dependency: svelte@4.0.2
node_modules/svelte
  peerOptional svelte@"^4.0.0" from ai@2.1.14
  node_modules/ai
    ai@"^2.1.14" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.```

**Comment by MaxLeiter** [comment]
This is an issue with the upstream SSWR library and our use of dependencies veruss optional deps. I'll take a look, thanks!

**Comment by MaxLeiter** [comment]
Closing as a duplicate of #248

**Comment by youngbloodcyb** [comment]
Thanks! And sorry, didn't see there was already an issue open 🤦

**Comment by nauticuff** [comment]
What I did to make this work on npm is Ctrl + Shift + F and find all svelte dependencies in my project and remove them from the package json files (I'm using React so no need for those). I have a working chatbot in my repository that is as bare bones as I could make it. Interestingly enough having some Vue dependencies gives no errors.
![nosvelte](https://github.com/vercel-labs/ai/assets/106128827/0ea3ff4f-6f70-4b5c-8dad-a2e4cd2eb471)

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This reports a concrete package installation failure caused by conflicting dependencies in the existing AI SDK package, so it is a bug despite being a duplicate.

### Evidence
- Installing ai@2.1.14 fails because its optional Svelte 4 peer conflicts with sswr@1.10.0's Svelte 3 peer requirement.
- A maintainer confirmed the problem stems from SSWR and the SDK's handling of dependencies versus optional dependencies.
- The issue was closed only because it duplicated #248, not because the reported behavior was invalid.
